pkg/manifests/manifests.go

```diff
@@ -1746,6 +1746,10 @@ func (f *Factory) PrometheusUserWorkload(grpcTLS *v1.Secret) (*monv1.Prometheus,
 		p.Spec.Retention = f.config.UserWorkloadConfiguration.Prometheus.Retention
 	}
 
+	if f.config.UserWorkloadConfiguration.Prometheus.RetentionSize != "" && f.config.UserWorkloadConfiguration.Prometheus.RetentionSize != "0" {
```

Comment:
I think it may be better to trim before comparing strings.

Reply:
I've put a comment on the other PR #1579: I'd be fine if we don't treat "0" as a special case.

> Isn't 0 an acceptable value?

it is a valid value and means the same as "".

> is "0GB" acceptable?

yes.

Reply:
ya 0 is valid for [parse2bytes](https://github.com/alecthomas/units/blob/b94a6e3cc13755c0a75fffecbb089eb346fc4289/bytes.go#L31-L37)  used by PO validate function , there was sort of confusion when `0` was specified to avoid in jira. I have updated pr https://github.com/openshift/cluster-monitoring-operator/pull/1579 will update this one too :)